python/hail/methods/misc.py

```diff
@@ -307,3 +307,53 @@ def wrap_input(interval):
     intervals = [wrap_input(x)._jrep for x in intervals.value]
     jmt = Env.hail().methods.FilterIntervals.apply(ds._jvds, intervals, keep)
     return MatrixTable(jmt)
+
+@typecheck(mt=MatrixTable, base_pairs=int)
+def window_by_locus(mt: MatrixTable, base_pairs: int) -> MatrixTable:
+    """Collect arrays of row and entry values from previous variants.
+
+    .. include:: ../_templates/req_tvariant.rst
+
+    .. include:: ../_templates/experimental.rst
+
+    Examples
+    --------
+    >>> ds_result = hl.window_by_locus(ds, 3)
+
+    Notes
+    -----
+    This method groups each row (variant) with the previous rows in a window of
+    `base_pairs` base pairs, putting the row values from the previous variants
+    into `prev_rows` (row field of type ``array<struct>``) and entry values
+    from those variants into `prev_entries` (entry field of type
+    ``array<struct>``).
+
+    The base_pairs argument is inclusive; if `base_pairs` is 2 and the loci are
+
+    .. code-block:: text
+
+        1:100
+        1:100
+        1:102
+        1:102
+        1:103
+        2:100
+        2:101
+
+    Then the size of `prev_rows` is 0, 1, 2, 3, 2, 0, and 1, respectively (and
+    same for the size of prev_entries). The `base_pairs` argument is inclusive;
+    if `base_pairs` is ``3``, then the locus ``1:100`` will be collected into
+    `prev_rows` for ``1:101``, ``1:102``, and ``1:103``.
+
+    Parameters
+    ----------
+    mt : :class:`.MatrixTable`
+        Input dataset.
+    base_pairs : :obj:`int`
+        Base pairs to include in the backwards window (inclusive).
```

Comment:
although it's longer, I think clearer for name to be `base_pair_window_size` or `bp_window_size`. This isn't the "base pairs to include", but rather the size of the backwards window in base pairs.  If it's `3` than your window has size `3`.